gtars/src/uniwig/mod.rs

```diff
@@ -1,16 +1,722 @@
+    // in any way
+
+    Some((ctg.parse().unwrap(), st, en))
+
+}
+
+
+pub fn run_uniwig(matches: &ArgMatches) {
+    //println!("I am running. Here are the arguments: {:?}", matches);
+
+    let combinedbedpath = matches
+        .get_one::<String>("bed")
+        .expect("combined bed path is required");
+
+    let chromsizerefpath = matches
+        .get_one::<String>("chromref")
+        .expect("chromref path path is required");
+
+    let bwfileheader = matches
+        .get_one::<String>("fileheader")
+        .expect("fileheader is required");
+
+    let smoothsize = matches
+        .get_one::<i32>("smoothsize")
+        .expect("smoothsize required");
+
+    let output_type = matches
+        .get_one::<String>("outputtype")
+        .expect("output type is required");
+
+
+    uniwig_main(*smoothsize, combinedbedpath, chromsizerefpath, bwfileheader, output_type)
+
+
+}
+
+fn clamped_start_position(start:i32, smoothsize: i32) -> i32{
+    // This is for ensuring that the start position for every wiggle file is at a minimum equal to `1`
+    std::cmp::max(1, start - smoothsize)
+
+}
+
+pub fn uniwig_main(smoothsize:i32, combinedbedpath: &str, _chromsizerefpath: &String, bwfileheader: &str, output_type: &str){
+    // Main Function
+
+    //println!("Hello from Uniwig main");
+
+    let stepsize = 1;
+
+    // Set up output file names
+
+    let mut file_names: [String; 3] = ["placeholder1".to_owned(), "placeholder2".to_owned(), "placeholder3".to_owned()];
+    let mut meta_data_file_names: [String; 3] = ["placeholder1".to_owned(), "placeholder2".to_owned(), "placeholder3".to_owned()];
+
+    // TODO determine potential file types
+    file_names[0] = format!("{}_{}.{}", bwfileheader, "start", output_type);
+    file_names[1] = format!("{}_{}.{}", bwfileheader, "end", output_type);
+    file_names[2] = format!("{}_{}.{}", bwfileheader, "core", output_type);
+
+    meta_data_file_names[0] = format!("{}{}.{}", bwfileheader, "start","meta");
+    meta_data_file_names[1] = format!("{}{}.{}", bwfileheader, "end","meta");
+    meta_data_file_names[2] = format!("{}{}.{}", bwfileheader, "core","meta");
+
+
+
+    let chrom_sizes = match read_chromosome_sizes(combinedbedpath) {
+        // original program gets chromosome size from a .sizes file, e.g. chr1 248956422
+        // the original program simply pushes 0's until the end of the chromosome length and writes these to file.
+        // can we instead just use the last endsite for each chromosome to save space in th wiggle file?
+
+        Ok(chrom_sizes) => chrom_sizes,
+        Err(err) => {
+            println!("Error reading chromosome sizes: {}", err);
+            return; // Exit the main function on error
+        }
+    };
+
+
+
+
+
+    let chromosomes: Vec<Chromosome> = read_bed_vec(combinedbedpath);
+
+    let num_chromosomes = chromosomes.len();
+
+    println!(" DEBUG Number of Chromosomes{:?}", num_chromosomes);
```

Comment:
remove?

Reply:
Maybe we could make use of a library like this: https://github.com/tokio-rs/tracing